frontend/test/__support__/server-mocks/collection.ts

```diff
@@ -31,22 +32,73 @@ export function setupCollectionsEndpoints({
       query: { tree: true, "exclude-archived": true },
       overwriteRoutes: false,
     },
-    collections.filter(collection => !collection.archived),
+    buildCollectionTree(collections.filter(collection => !collection.archived)),
   );
   fetchMock.get(
     {
       url: "path:/api/collection/tree",
       query: { tree: true },
       overwriteRoutes: false,
     },
-    collections,
+    buildCollectionTree(collections),
   );
   fetchMock.get(
     { url: "path:/api/collection", overwriteRoutes: false },
     collections,
   );
 }
 
+function buildCollectionTree(collections: Collection[]) {
```

Comment:
This is a bit convoluted, but basically we want to build the tree structure of the collection to resemble the API response from `/api/collection/tree`. This is needed for testing `frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.tsx`